### PR TITLE
Concept for base task

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/code-editor-with-actions-behavior.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<script>
+CodeEditorWithActionsBehavior = {
+	selected: function(){
+		console.log('cdoe editor selected')
```
Comment (lorenzleutgeb) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/code-editor-with-actions-behavior.html]: d

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/code-editor-with-actions-behavior.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	},
+	deselected: function(){
+		console.log('cdoe editor deselected')
```
Comment (lorenzleutgeb) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/code-editor-with-actions-behavior.html]: d

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/input-view-behavior.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<script>
+InputViewBehavior = {
+	selected: function(){
+		console.log('input view selected')
```
Comment (lorenzleutgeb) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/input-view-behavior.html]: d

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/input-view-behavior.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	},
+	deselected: function(){
+		console.log('input view deselected')
```
Comment (lorenzleutgeb) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/input-view-behavior.html]: d

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/input-view-behavior.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+	},
+	_addPropertiesTo: function(element){
+		element.task = this.task;
+		return element;
+	},
+	_extractData: function(parent){
+		this.task = parent.task;
+	},
+	process: function(eventName, data){
+		switch(eventName) {
```
Comment (lorenzleutgeb) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/input-view-behavior.html]: `switch` with only one `case`? U mad?

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/tabbed-console-behavior.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<script>
+TabbedConsoleBehavior = {
+	selected: function(){
+		console.log('tabbed console selected')
```
Comment (lorenzleutgeb) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/tabbed-console-behavior.html]: d

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/tabbed-console-behavior.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	},
+	deselected: function(){
+		console.log('tabbed console deselected')
```
Comment (lorenzleutgeb) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/tabbed-console-behavior.html]: d

File: app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/challenge-leaderboard-behavior.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<script>
+LeaderboardBehavior = {
+	selected: function(){
+		var self = this;
+		self._refresh();
+		this.intervalId = setInterval(function(){
+			self._refresh();
+		}, 5000);
```
Comment (victorbalan) [inline app/elements/coder/challenge/tasks/endpoints/base-components/behaviors/challenge-leaderboard-behavior.html:9]: this is what @pbochis coded so the leaderboard refreshes once every 5 seconds

**Pull request merged**